### ISSUE TITLE
Default `vcs:` and some `gitHub:` tags missing from newly created stack.

### ISSUE THREAD

**Issue by wtfzambo** [pulumi/pulumi#12616]
### What happened?

Working on a pulumi project on the default stack (`dev`), I later created a new one (`iam`), added some code, and then ran `pulumi up`.

Opening the console, I noticed that the newly created stack was missing the repo info (repo name would be `glue-on-pulumi`):

<img src="https://user-images.githubusercontent.com/47462792/230420933-180205d0-992d-4432-aa44-40b22525b7cd.png" width=40%>

Running `pulumi stack tag ls`, I notice that the new stack indeed does not have the `vcs:` and `gitHub:repo` tags set:

<img src="https://user-images.githubusercontent.com/47462792/230421626-f68f1d43-5f51-4204-bc61-3a6dda8eed98.png" width=50%>

### Expected Behavior

I expect that new stacks created in a project under a git repo to also include the various `vcs:` and `gitHub:repo` tags set.

### Steps to reproduce

I'm running this in a devcontainer, using Docker Desktop and WSL2 (Ubuntu 20.04) backend and filesystem.

You can find the devcontainer template [here](https://github.com/wtfzambo/glue-devcontainer-template), but it's missing pulumi so it needs to be installed manually, if you decide to test using such devcontainer.

Steps:
- Be in a pulumi project in a git repository
- Create a new stack
- Run `pulumi up`
- Run `pulumi stack tag ls`
- Check that `vcs:` tags are missing

### Output of `pulumi about`

<details>

<summary>Pulumi about</summary>

```
CLI          
Version      3.59.1
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME    VERSION
aws     5.33.0
python  unknown

Host     
OS       amazon
Version  2
Arch     x86_64

This project is written in python: executable='/usr/local/bin/python3' version='3.10.2
'

Current Stack: <ORG>/glue-jobs/temp

TYPE                              URN
pulumi:providers:aws              urn:pulumi:temp::glue-jobs::pulumi:providers:aws::default
pulumi:pulumi:Stack               urn:pulumi:temp::glue-jobs::pulumi:pulumi:Stack::glue-jobs-temp
pulumi:providers:aws              urn:pulumi:temp::glue-jobs::pulumi:providers:aws::default_5_33_0
pulumi:providers:pulumi           urn:pulumi:temp::glue-jobs::pulumi:providers:pulumi::default
aws:s3/bucketObject:BucketObject  urn:pulumi:temp::glue-jobs::aws:s3/bucketObject:BucketObject::slow_changing_dimensions/utils_.py
aws:s3/bucketObject:BucketObject  urn:pulumi:temp::glue-jobs::aws:s3/bucketObject:BucketObject::slow_changing_dimensions/scd.py
pulumi:pulumi:StackReference      urn:pulumi:temp::glue-jobs::pulumi:pulumi:StackReference::<ORG>/glue-jobs/iam
aws:glue/job:Job                  urn:pulumi:temp::glue-jobs::aws:glue/job:Job::slow_changing_dimensions_temp


Found no pending operations associated with temp

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/<ORG>
User           <USER>
Organizations  <ORG>

Dependencies:
NAME            VERSION
chispa          0.9.2
flake8-black    0.3.6
glue-on-pulumi  0.1.0
isort           5.12.0
pip             23.0.1
pytest          7.2.2
python-dotenv   1.0.0
setuptools      67.4.0
wheel           0.38.4

Pulumi locates its logs in /tmp by default
```

</details>

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
> Be in a pulumi project in a git repository
> Create a new stack
> Run pulumi stack tag ls
> Check that vcs: tags are missing

So that's _expected_, we don't lookup and set the tags until we do an update. But it sounds like from your earlier description that in this new iam stack you did run `up` and the tags were still missing?

**Comment by wtfzambo** [comment]
> you did run `up` and the tags were still missing?

Yes sorry, I forgot to add that step. I did indeed run `pulumi up`, and nonetheless the tags didn't get set. I will update the description.

As a side note, I tested the same issue switching filesystem (so now the container mounts Windows files), but the same issue persists.

I thought it might be due to some UID / GID Linux shenanigans but that doesn't seem to be the case.

**Comment by lunaris** [comment]
Hi @wtfzambo; is this still an issue for you? I can't currently reproduce this and we've tried a couple of platforms including WSL. Since you filed this issue we have improved the behaviour to tag stacks at `init` time too, so perhaps it's no longer a problem? I'm going to close this as stale but if it's still affecting you please feel free to reopen/let us know! 🙏
